### ISSUE TITLE
Add cross-validation support for LSTM training

### ISSUE THREAD

**Issue by Acusick1** [Acusick1/odds#90]
**Goal**: Bring LSTM training to parity with XGBoost by implementing cross-validation and early stopping.

**Complexity**: 🟡 Medium

**Location**:
- `packages/odds-analytics/odds_analytics/lstm_strategy.py`

**Background**:
XGBoost has full CV support via `train_with_cv()` and `run_cv()`. LSTM currently:
- Has `patience`/`min_delta` in config but NOT implemented
- Returns only training loss (no validation metrics)
- No cross-validation support

**Scope**:

1. **Implement early stopping** in `train()`:
   - Track validation loss each epoch
   - Stop if no improvement for `patience` epochs
   - Use `min_delta` threshold from config
   - Return best model (not last epoch)

2. **Add validation metrics**:
   - Calculate val_mse, val_mae, val_r2 each epoch
   - Return in history dict (matching XGBoost format)

3. **Add `train_with_cv()` method**:
   - Reuse existing `run_cv()` infrastructure from cross_validation.py
   - Support both `timeseries` and `kfold` methods
   - Aggregate CV metrics (mean ± std)

4. **Update training history format**:
   ```python
   # Current
   {"loss": [epoch1, epoch2, ...]}

   # New (match XGBoost)
   {
       "train_mse": float,
       "train_mae": float,
       "val_mse": float,
       "val_mae": float,
       "val_r2": float,
       # Plus CV metrics if using CV
   }
   ```

**Success criteria**:
- [ ] Early stopping implemented using patience/min_delta config
- [ ] Validation metrics tracked per epoch
- [ ] `train_with_cv()` works with TimeSeriesSplit
- [ ] History format matches XGBoost for consistency
- [ ] CLI `odds train run` works with LSTM + CV
- [ ] Pruning integration ready for HPO (reports intermediate values)

**Prerequisites**:
- None (can be done independently)

**Related**:
- #81 - HPO needs intermediate value reporting for pruning

**Related files**:
- `packages/odds-analytics/odds_analytics/lstm_strategy.py` - LSTM training
- `packages/odds-analytics/odds_analytics/training/cross_validation.py` - CV infrastructure
- `packages/odds-analytics/odds_analytics/training/config.py` - LSTMConfig with patience/min_delta
